docs/references/architecture/adr-119-dog-mempool-gossip.md

```diff
@@ -0,0 +1,287 @@
+via a new config flag `enable_dog_protocol`. More details on this in the sections below. 
+
+## Detailed Design
+
+We start the design section with a general description of how the algorithm works. 
+
+CometBFT caches received transactions and stores the IDs of all the peers that have sent it in a `senders` list. When a transaction enters the mempool,
+if it was already received it can be retrieved from the cache. When this happens a certain number of times, the receiving node notifies the sender to stop forwarding any transaction from the peer 
+that had originally sent the transaction. The exact point when the notification is sent is defined below as a redundancy threshold. (TODO LiNK)
+
+Namely, let's assume that transaction `tx_1` was sent by node `B` to node `C` which forwarded it to `A`. `A` already has the transaction and notifies node `C` about this. Node `C` will look up the senders of transaction `tx_1` and in the future disable transaction forwarding from node `B` to node `A`.
+
+In reality, if node `C` ha received `tx_1` from multiple senders (`X`,`Y`, `B`), it will pick the first one that has sent it and disable that route. 
+
+Node `C` keeps a map of `disabledRoutes` per peer and uses it to determine whether a transaction should be forwarded. 
+
+Entries for a particular peer are reset if a peer sends a `Reset` message or if a peer disconnects. 
+
+### Redundancy control
+
+In an ideal setting, receiving a transaction more than once is not needed. But in a Byzantine setting, allowing for only one route for a transaction can introduce an attack vector. 
+
+Operators can thus configure a desired redundancy and the gossip protocol will adjust route disabling based on this setting. Redundancy is defined as the ratio between duplicate and unique transactions. 
+
+<!-- Operators can also define how much they tolerate the redundancy to deviate from the desired level. -->
+
+By default, the redundancy is set to `1` and impacts the gossiping of transactions as follows:
+
+
+```go
+  redundancy := duplicateTxs / firstTimeTxs
+  if redundancy < r.redundancyLowerBound:
+    peer.send(Reset)
+  if redundancy > redundancyUpperBound
+    enableRouteDisabling() 
+```
+
+The number of unique and duplicate transactions is upadated as transactions come in and,
+periodically (every `1s`), DOG recomputes the redundnacy of the system. 
+
+
+#### Redundancy adjustment triggering
+
+As explained, redundancy is adjusted periodicially. It is also explicitly triggered when a peer is disconnected. <!-- TODO justify why-->
+
+### Impacted areas of code
+
+The bulk of the changes is constrained to the mempool reactor. 
+
+
+#### Breaking changes and new additions
+
+- The `Mempool` interface is expanded with a method : 
+```go
+// GetSenders returns the list of node IDs from which we receive the given transaction.
+GetSenders(txKey types.TxKey) ([]nodekey.ID, error)
```

Comment:
```suggestion
GetSenders(txKey types.TxKey) ([]p2p.ID, error)
```

Reply:
Ok noted, we are keeping it in line with main until it gets reverted but note to change it.